### ISSUE TITLE
Portable apps using Microsoft.Win32.Registry fail on non-Windows

### ISSUE THREAD

**Issue by eerhardt** [dotnet/core-setup#2377]
## Steps to reproduce
Using latest 2.0.0-preview2 SDK and SharedFx

dotnet new console
Add to your project
```
  <ItemGroup>
  <PackageReference Include="Microsoft.Win32.Registry" Version="4.4.0-*" />
  </ItemGroup>
```
Write some code using the registry like so:
```
            RegistryKey regKeyStorageKey = null;
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                regKeyStorageKey = Registry.CurrentUser.CreateSubKey("Test9999");
            }
            if (regKeyStorageKey != null)
            {
                // do something with registry
            }
```
dotnet publish
Take the program to a linux machine and try to run it.

## Expected  behavior
The app runs successfully.

## Actual behavior
```
Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Win32.Registry, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

## Environment data
```
F:\dotnet\dotnet.exe --info
.NET Command Line Tools (2.0.0-preview1-005973)

Product Information:
 Version:            2.0.0-preview1-005973
 Commit SHA-1 hash:  5436217013

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Users\eerhardt\.dotnet\x64\sdk\2.0.0-preview1-005973\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview2-25309-07
  Build    : 41f5fc94eedc889f086800c23f35bf14a8c75a9f
```

**Comment by eerhardt** [comment]
It appears the problem is that https://github.com/dotnet/corefx/commit/c5db9a803ee1eb2c9c33f8101a3d1692c0158317 removed `Microsoft.Win32.Registry` from the Shared Framework on non-Windows.  However, publish respects what is in the `Microsoft.NETCore.App.PlatformManifest.txt` file, and if a file shows up there, it is excluded from the published output.

So we either need to put these assemblies back in all Shared Frameworks, or remove it completely from the manifest file.

/cc @ericstj @weshaggard

**Comment by weshaggard** [comment]
@ericstj do we want to have the manifiest account for this or update the shared frameworks to be the same?

**Comment by ericstj** [comment]
>  do we want to have the manifiest account for this

I don't think so.  We need those entries in the manifest to prevent someone from referencing an older version of the package and downgrading the file in the shared framework.

We should add it back to all frameworks.  I think that'll have to be a requirement.

**Comment by weshaggard** [comment]
OK I'm reverting my change to optimize some shared frameworks https://github.com/dotnet/corefx/pull/19780.

**Comment by eerhardt** [comment]
This is now fixed with the latest 2.0.0-preview2 build.  Thanks @weshaggard.
